bangazon-clumsy-penguins/BangazonAPI
# Eh training programs

# Name - Training Programs

## Description
##### Provide clarity to any potentially complex code:
Created TrainingsController to allow developers to access Trainings table

##### Explain reasons behind organizational or architectual decisions you made:

## Steps to Test

- [ ] FIRST add a few Employees to each Training.
THEN run a GET request to verify that all Trainings are returned and that each Training includes all registered Employees.

- [ ] FIRST add a few Trainings with EndDates that are today and in the future.
THEN run a GET request with `?completed=false` in the query to verify that the only Trainings returned are ones that do not have an EndDate in the past

- [ ] Run a GET request with an Id in the route to verify that a single Training object is returned and that it includes all registered Employees

- [ ] Run a GET request with an invalid Id in the route to verify that a "404 - Not Found" exception is thrown

- [ ] Run a POST request to verify that a Training object is added to the database

- [ ] Run a POST request for a Training that has a non-positive MaxOccupancy to verify that an exception is thrown ("The property 'MaxOccupancy' must be greater than 0; cannot create a training with zero or negative occupants")

- [ ] Run a PUT request with the Id of a current training to verify that it is edited in the database

- [ ] Run a PUT request with the Id of a current training and a non-positive MaxOccupancy to verify that an exception is thrown ("The property 'MaxOccupancy' must be greater than 0; cannot create a training with zero or negative occupants")

- [ ]  FIRST add at least one training with a StartDate in the future.
THEN run a DELETE request with the Id of a training that starts in the future
THEN run a GET request to verify that it was removed from the database

- [ ] Run a DELETE request with the Id of a training that starts today or in the past to verify that an exception is thrown ("The property StartDate must be in the future; cannot delete a training that has already started")

##### System configuration.
None

##### 3rd party libraries that need to be installed.
None

##### Command line utilities to run.
None

##### Link to Feature Ticket
#9
